front/src/modules/timetableItem/hooks/useSelectedTimetableItem.tsx

```diff
@@ -0,0 +1,28 @@
+import { useMemo } from 'react';
+
+import { useSelector } from 'react-redux';
+
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import type { TimetableItem } from 'reducers/osrdconf/types';
+import { getSelectedTrainId } from 'reducers/simulationResults/selectors';
+import { extractPacedTrainIdFromOccurrenceId, isOccurrenceId } from 'utils/trainId';
+
+const useSelectedTimetableItem = (): TimetableItem | undefined => {
+  const trainId = useSelector(getSelectedTrainId);
+
+  const timetableItemId = useMemo(() => {
```

Comment:
Nit: we could probably drop useMemo here. The logic is fast and depends only on trainId, so memoizing it isn’t really needed, i think.

Reply:
Done, I also created a small function to group the logic